lib/dialects/abstract/utils.js

```diff
@@ -0,0 +1,37 @@
+
+const _dialects = {};
+
+/**
+ * Register dialect implementation.
+ *
+ * @param {string}  [name] the name of dialect.
+ * @param {AbstractDialect} [dialectClass] dialect implemetation as subclass of AbstractDialect.
+ */
+function registerDialectImpl( name, dialectClass ) {
```

Comment:
I would recommend removing the `name` parameter to, and instead add it as a static property of `AbstractDialect`:

```typescript
class Db2Dialect extends AbstractDialect {
  static getDialectName() {
    return 'db2';
  }
}
```

this way users can simply do

```
Sequelize.registerDialect(MyCustomDialect);
```

without specifying the name themselves (this is related to the comment regarding side effects in imports)